setup.py

```diff
@@ -384,6 +385,22 @@ def run(self):
     long_description = f.read().split('"""')[1]
 
 if __name__ == '__main__':
+    # Fetch git commit hash and write down to commit_hash.txt before
+    # shipped in tarball.
+    commit_hash = None
+    commit_hash_filepath = 'doc/commit_hash.txt'
```

Comment:
This should only be done as part of the sdist step.

Reply:
This file gets written to the repo, meaning if one later does `cd doc; make html` it will use that instead of the rev-parse.